### ISSUE TITLE
How to stop the background running Analyze data for oracle? Because I only use one username, but it analyze all users of the database.

### ISSUE THREAD

**Issue by gclsoft** [metabase/metabase#8955]
```
11-16 21:57:59 DEBUG sync-metadata.fields :: Hash of '表单2,371“DBMYLIKE.T_MED_OLDCASEHISTORY”' matches stored hash, skipping fields sync for table
11-16 21:57:59 DEBUG sync-metadata.fields :: Hash of '表单2,372“SYSMAN.MGMT_CORRECTIVE_ACTION”' matches stored hash, skipping fields sync for table
11-16 21:57:59 DEBUG sync-metadata.fields :: Hash of '表单2,373“DBMYLIKE.V_REC_RECEIVE_EXTENSION_AMT”' matches stored hash, skipping fields sync for table
11-16 21:57:59 DEBUG sync-metadata.fields :: Hash of '表单2,374“APEX_030200.WWV_FLOW_USER_ACTIVITY_LOG”' matches stored hash, skipping fields sync for table
11-16 21:58:00 DEBUG sync-metadata.fields :: Hash of '表单2,375“SYSMAN.MGMT_CURRENT_VIOLATION”' matches stored hash, skipping fields sync for table
11-16 21:58:00 DEBUG sync-metadata.fields :: Hash of '表单2,376“SYSMAN.ECM$HIST_ORACLE_SOFTWARE”' matches stored hash, skipping fields sync for table
11-16 21:58:00 DEBUG sync-metadata.fields :: Hash of '表单2,377“SYSMAN.MGMT$STORAGE_REPORT_PATHS”' matches stored hash, skipping fields sync for table
11-16 21:58:00 DEBUG sync-metadata.fields :: Hash of '表单2,378“SYSMAN.MGMT_MP_DEPLOYMENTS”' matches stored hash, skipping fields sync for table
11-16 21:58:00 DEBUG sync-metadata.fields :: Hash of '表单2,379“SYSMAN.MGMT_AVAILABILITY_MARKER”' matches stored hash, skipping fields sync for table
11-16 21:58:00 DEBUG sync-metadata.fields :: Hash of '表单2,380“DBMYLIKE.T_APP_POSTER”' matches stored hash, skipping fields sync for table
11-16 21:58:01 DEBUG sync-metadata.fields :: Hash of '表单2,381“DBMYLIKE.CT_CRM_WECHAT_INTENT”' matches stored hash, skipping fields sync for table
11-16 21:58:01 DEBUG sync-metadata.fields :: Hash of '表单2,382“APEX_030200.WWV_FLOW_USER_ACCESS_LOG_V”' matches stored hash, skipping fields sync for table
11-16 21:58:01 DEBUG sync-metadata.fields :: Hash of '表单2,383“SYSMAN.MGMT$CPF_PATCH_INFO”' matches stored hash, skipping fields sync for table
11-16 21:58:02 DEBUG sync-metadata.fields :: Hash of '表单2,384“DBMYLIKE.T_ORG_ADMIN”' matches stored hash, skipping fields sync for table
11-16 22:03:22 INFO sync.util :: FINISHED: Analyze data for oracle数据库2“dev3” (52 s)
```
How to stop the background running Analyze data for oracle? Because I only use the username of DBMYLIKE, but it analyze all users of the database.

**Comment by gclsoft** [comment]
How to stop the other users, such as `SYSMAN`,`USERNAME_BK` to analyze data, because I only need one username.

**Comment by gclsoft** [comment]
![image](https://user-images.githubusercontent.com/1269898/48712036-be0a9d80-ec47-11e8-9684-46197116a7a6.png)
![image](https://user-images.githubusercontent.com/1269898/48712113-eb574b80-ec47-11e8-86b6-79293b1004d0.png)

How to only use one username, don't list so much,I connect the database by one username is enough. Too many username makes problems:
`Possible timezone conflict found on database dev3. JVM timezone is Asia/Shanghai and detected database timezone is America/Chicago. Configure a report timezone to ensure proper date and time conversions.
11-19 18:51:55 INFO sync.analyze :: fingerprint-fields Analyzed [*****·············································]    10% Table 931 'SYSMAN.ECM$HIST_OS_REGISTERED_SW'
11-19 18:52:10 INFO sync.analyze :: fingerprint-fields Analyzed [**********········································]    20% Table 1,858 'SYSMAN.MGMT$SOFTWARE_COMP_PATCHSET'
11-19 18:52:15 ERROR sync.util :: Error running sync step: Multiple methods in multimethod 'fingerprinter' match dispatch value: [:type/Date :type/FK] -> [:type/DateTime :type/*] and [:type/* :type/FK], and neither is preferred`

**Comment by camsaul** [comment]
I think Oracle treats usernames as the equivalent of schemas so I think the actual problem here is that it is syncing schemas in your database that you don't want it to sync.

Metabase will go ahead and sync all the schemas it has access to. If you don't want to sync those schemas it might be easiest to create a new User and restrict it to only be allowed to access the schemas you want to sync. Otherwise there's not currently a way to restrict sync to only a certain schema or set of schemas.

**Comment by beders** [comment]
Metabase is unusable for Oracle without the ability to exclude schemas

**Comment by lindsay-stevens** [comment]
Seems like a subset or duplicate of #5500

**Comment by doubler84** [comment]
I think that for Oracle is very important this feature.
I would like to choose Oracle schema that metabase have to sync.

**Comment by doricardo** [comment]
Is there a solution to show only on schema?

**Comment by flamber** [comment]
@doricardo 
Yes, create a user in Oracle that only has access to the schemas you want - https://github.com/metabase/metabase/issues/8955#issuecomment-460888182

**Comment by beders** [comment]
That is not a workable solution as our users don’t have admin rights.

**Comment by vvaezian** [comment]
It was mentioned above `"there's not currently a way to restrict sync to only a certain schema or set of schemas"`. I assume this is referring to restricting on db side. But can't the restriction be done on Metabase side instead? i.e. it would fetch all schemas, but ignores the ones that are in "All Except.." list?

<img src="https://github.com/user-attachments/assets/108c9fe6-8de7-40c8-b292-d78282e0242b" width="400px"></img>

There are hundreds of tables (and thousands of fields) in these "system schemas". When I first connected my db it took a long time to finish the initial sync.

<img src="https://github.com/user-attachments/assets/d4df2224-f996-4e01-bc18-e9d164703057" height="400px"></img>

<img src="https://github.com/user-attachments/assets/12ae598a-4c88-4a9d-9c51-d29d95eb04e9" height="400px"></img>

**Comment by ranquild** [comment]
We should implement the schema filter feature as for postgres 

<img src="https://uploads.linear.app/a24a1a82-ddc8-4567-aacc-c3804ea2385e/8acff9a9-d955-4ffc-8b08-8efe98a9d0f3/f01d5969-9d19-429c-b338-8351a6fa9f4d?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2EyNGExYTgyLWRkYzgtNDU2Ny1hYWNjLWMzODA0ZWEyMzg1ZS84YWNmZjlhOS1kOTU1LTRmZmMtOGIwOC04ZWZlOThhOWQwZjMvZjAxZDU5NjktOWQxOS00MjljLWIzMzgtODM1MWE2ZmE5ZjRkIiwiaWF0IjoxNzQxMzU5MTQ0LCJleHAiOjMzMzExOTE5MTQ0fQ.h0MqmAt05Tjvz1-q3QVGuBLMvg3b7GWChQRyo2JPBqk " alt="Screenshot 2025-03-07 at 09.47.58.png" width="730" data-linear-height="220" />
